test/python/utils/message.py

```diff
@@ -168,6 +169,46 @@ def raw(self) -> bytes:
         raw += self.block_payload_hash
         return raw
 
+class AttestationDal(Message):
+    """Class representing an attestation + DAL."""
+    chain_id: int
+    branch: bytes
+    slot: int
+    op_level: int
+    op_round: int
+    block_payload_hash: bytes
+    dal_message: bytes
+
+    def __init__(self,
+                 chain_id: int,
+                 branch: Union[str,bytes],
+                 slot: int,
+                 op_level: int,
+                 op_round: int,
+                 block_payload_hash: Union[str,bytes],
+                 dal_message: Union[str,bytes]):
+        self.chain_id = chain_id
+        self.branch = scrub(branch)
+        assert_data_size("branch", self.branch, 32)
+        self.slot = slot
+        self.op_level = op_level
+        self.op_round = op_round
+        self.block_payload_hash = scrub(block_payload_hash)
+        assert_data_size("block_payload_hash", self.block_payload_hash, 32)
+        self.dal_message = scrub(dal_message)
+
+    def raw(self) -> bytes:
+        raw = b''
+        raw += MagicByte.TENDERBAKE_ENDORSEMENT.to_bytes(1, byteorder='big')
+        raw += self.chain_id.to_bytes(4, byteorder='big')
+        raw += self.branch
+        raw += OperationTag.ATTESTATION_WITH_DAL.to_bytes(1, byteorder='big')
+        raw += self.slot.to_bytes(2, byteorder='big')
+        raw += self.op_level.to_bytes(4, byteorder='big')
+        raw += self.op_round.to_bytes(4, byteorder='big')
+        raw += self.block_payload_hash
+        return raw
```

Comment:
The dal-endorsements have a `dal_attestation` field in addition to those for endorsements.

In `tezos/src/proto_alpha/lib_protocol/operation_repr.ml`:
```ocaml
  let consensus_content_encoding =
    obj4
      (req "slot" Slot_repr.encoding)
      (req "level" Raw_level_repr.encoding)
      (req "round" Round_repr.encoding)
      (req "block_payload_hash" Block_payload_hash.encoding)

  let dal_content_encoding =
    obj1 (req "dal_attestation" Dal_attestation_repr.encoding)

  let endorsement_encoding = consensus_content_encoding

  let endorsement_with_dal_encoding =
    merge_objs consensus_content_encoding dal_content_encoding
```

You may just have forgotten to add the `self.dal_message`.

Reply:
Done